File: solutions/1765/main.py
from collections import deque
from typing import List

class Solution:
    def highestPeak(self, isWater: List[List[int]]) -> List[List[int]]:
        # initialize
        row, col = len(isWater), len(isWater[0])
        ans = [[-1] * col for _ in range(row)]
        que = deque(maxlen=row*col)
        dirs = [(1,0), (-1,0), (0,1), (0,-1)]
        # push water cells in queue
        for i in range(row):
            for j in range(col):
                if isWater[i][j] == 1:
                    que.append((i, j))
                    ans[i][j] = 0
        # bfs
        h = 1
        while len(que) > 0:
            q_size = len(que)
            while q_size > 0:
                x, y = que.popleft()
                for d in dirs:
                    nx, ny = x+d[0], y+d[1]
                    if nx < 0 or nx >= row or ny < 0 or ny >= col:
                        continue
                    if ans[nx][ny] == -1:
                        ans[nx][ny] = h
                        que.append((nx, ny))
                q_size -= 1
            h += 1
        return ans

if __name__ == "__main__": 
    isWater = [[0,1],[0,0]]
    s = Solution()
    print(s.highestPeak(isWater))

File: solutions/1765/main.rs
use std::collections::VecDeque;

struct Solution {}

impl Solution {
    pub fn highest_peak(is_water: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        // initialize 
        let row = is_water.len(); 
        let col = is_water[0].len(); 
        let mut ans = is_water.clone(); 
        let mut que = VecDeque::new();
        let dirs: Vec::<(i32, i32)> = vec![(1,0), (-1,0), (0,1), (0,-1)]; 
        // put water cells in queue 
        for i in 0..row {
            for j in 0..col {
                if is_water[i][j] == 1 {
                    ans[i][j] = 0; 
                    que.push_back((i, j)); 
                }
                else {
                    ans[i][j] = -1;
                }
            }
        }
        // bfs 
        let mut h: i32 = 1; 
        while !que.is_empty() {
            let mut q_size = que.len(); 
            while q_size > 0 {
                let (x, y) = que.pop_front().unwrap(); 
                for d in &dirs {
                    let (nx, ny) = ((x as i32+d.0) as usize, (y as i32+d.1) as usize); 
                    if nx >= row || ny >= col {
                        continue; 
                    }
                    if ans[nx][ny] == -1 {
                        ans[nx][ny] = h; 
                        que.push_back((nx, ny)); 
                    }
                }
                q_size -= 1; 
            }
            h += 1; 
        }
        return ans; 
    }
}

fn main() {
    let is_water = vec![vec![0,1], vec![0,0]];
    let result = Solution::highest_peak(is_water);
    println!("{:?}", result);
}
